Xamarin.Forms.Platform.Android/AppCompat/FormsAppCompatActivity.cs

```diff
@@ -115,26 +114,17 @@ protected void LoadApplication(Application application)
 			}
 
 			_application = application ?? throw new ArgumentNullException(nameof(application));
-			(application as IApplicationController)?.SetAppIndexingProvider(new AndroidAppIndexProvider(this));
+			(application as IApplicationController).SetAppIndexingProvider(new AndroidAppIndexProvider(this));
 			Xamarin.Forms.Application.SetCurrentApplication(application);
 
 			if (Xamarin.Forms.Application.Current.OnThisPlatform().GetWindowSoftInputModeAdjust() != WindowSoftInputModeAdjust.Unspecified)
 				SetSoftInputMode();
 
 			CheckForAppLink(Intent);
 
+			application.PropertyChanged -= AppOnPropertyChanged;
 			application.PropertyChanged += AppOnPropertyChanged;
 
-			if (application?.MainPage != null)
```

Comment:
I think we still need to dispose the old main page renderer, on `_application`, before L116

Reply:
Good point, I tested with the same application object for the two calls but I also need to test with a different application object.
I will test that case.

Reply:
I tested the two cases, with the same application or a different it work fine in all case.

I tried to dispose the renderer on _application but it's too early.

With dispose + clear :
![image](https://user-images.githubusercontent.com/1446221/49934396-8c939380-fece-11e8-88e5-6c871114a604.png)

With clear only :
![image](https://user-images.githubusercontent.com/1446221/49934382-81d8fe80-fece-11e8-9a42-0224f7476801.png)

That the same issue that the current code.

Reply:
I deployed the fix in the staging version of my application.
I dont have anymore report of NotSupportedException but I have now reports of NullReferenceException inside the layout method from some devices.

I did further investigations and found the origin of the layout call.
The layout is triggered by the affectation of the ToolbarVisibility [here](https://github.com/xamarin/Xamarin.Forms/blob/ceb143a2a5f970a70561c53c14f8c9a3328999c4/Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs#L136).
For testing I removed the call and now I didn't have the layout anymore time and can clear the renderer earlier whitout the NRE.

After that I tested the IsLayoutRequested property and it was true, so I added a check to this condition before request another layout pass.

It fix the issue in my reproduction code but I dont know if it fix all the cases.
This layout request call seems out of place compared to others.
What's your opinion on that ?

Reply:
Not sure if I am following, but could it be that you are disposing the same page? ie: is this the same application object with the same page?
Also, that last null exception - is that the old page that is throwing, or the new one? If it is the old page, then why is it trying to lay it out?

Reply:
Yes the dispose problem occurs because it's the same page with the same app.
The null reference is throwed because of the layout tirggered by setting the toolbar visibility and if the page have already been disposed like in this case, it crash.